### ISSUE TITLE
THREE.WebGLRenderer: image is not power of two (0x0). Resized to 0x0

### ISSUE THREAD

**Issue by Davidx7** [mistic100/Photo-Sphere-Viewer#7]
Hi

Trying to finalize the Angular plugin for PH, but I get this error.

Replicating:: 
When adding a 360 photo (taken with normal google camera) at a resolution ```10348x5174``` -- which is a power of 2. This version works.
Warning log:
```
THREE.WebGLRenderer: OES_texture_float_linear extension not supported.
pack.3d.js:10 THREE.WebGLRenderer: image is not power of two (4095x2047). Resized to 4096x2048 <canvas width=​"4095" height=​"2047">​
```

-- Fixed with proposed .min. setting.

```this.prop.size``` = ```size: Object
height: 850
image_height: 2047.9999999999998
image_width: 4095.9999999999995
ratio: 0.9035294117647059
width: 768```

When adding a 360 photo (taken with another phone) at a resolution of ```5376x2688``` -- which is a power of 2, doesn't work. -- image is black and cavans size turns to ```0```

Both are ```jpg```, both are with standard cameras, image formats are similar, can't really find the problem

This is error log::
```
THREE.WebGLRenderer: image is not power of two (0x0). Resized to 0x0 <canvas width=​"0" height=​"0">​
THREE.WebGLRenderer: Texture is not power of two. Texture.minFilter should be set to THREE.NearestFilter or THREE.LinearFilter. T…E.Texture {uuid: "F710BF59-6F15-4809-9606-48C0D99A4906", name: "", sourceFile: "", image: canvas, mipmaps: Array[0]…}_listeners: Objectdispose: Array[1]__proto__: Objectanisotropy: 1flipY: trueformat: 1021generateMipmaps: trueid: 2image: canvasmagFilter: 1006mapping: 300minFilter: 1008mipmaps: Array[0]name: ""offset: THREE.Vector2onUpdate: nullpremultiplyAlpha: falserepeat: THREE.Vector2sourceFile: ""type: 1009unpackAlignment: 4uuid: "F710BF59-6F15-4809-9606-48C0D99A4906"version: 1wrapS: 1001wrapT: 1001__proto__: THREE.Texture
WebGL: INVALID_VALUE: texImage2D: no canvas
index.html:1 [.WebGLRenderingContext-0xb811db40]RENDER WARNING: texture bound to texture unit 0 is not renderable. It maybe non-power-of-2 and have incompatible texture filtering.```

```this.prop.size``` = ```height: 850
image_height: NaN
image_width: NaN
ratio: 0.9035294117647059
width: 768```

The canvas with the markers appears, but the texture is black.
Each time I move around, it generates lots of times the warning 

```[.WebGLRenderingContext-0xb74c4a78]RENDER WARNING: texture bound to texture unit 0 is not renderable. It maybe non-power-of-2 and have incompatible texture filtering.```

**Comment by Davidx7** [comment]
Found the problem. It's a exif data problem

Your parsing method, parses this XML 
``` <rdf:Description rdf:about=""
        xmlns:GPano="http://ns.google.com/photos/1.0/panorama/"
      GPano:UsePanoramaViewer="True"
      GPano:ProjectionType="equirectangular"
      GPano:PoseHeadingDegrees="356.8"
      GPano:CroppedAreaLeftPixels="0"
      GPano:FullPanoWidthPixels="10348"
      GPano:FirstPhotoDate="2015-12-17T20:21:46.760Z"
      GPano:CroppedAreaImageHeightPixels="5174"
      GPano:FullPanoHeightPixels="5174"
      GPano:SourcePhotosCount="64"
      GPano:CroppedAreaImageWidthPixels="10348"
      GPano:LastPhotoDate="2015-12-17T20:24:39.711Z"
      GPano:CroppedAreaTopPixels="0"
      GPano:LargestValidInteriorRectLeft="0"
      GPano:LargestValidInteriorRectTop="0"
      GPano:LargestValidInteriorRectWidth="10348"
      GPano:LargestValidInteriorRectHeight="5174"/>``` and does it correctly. Some cameras. 

Other cameras structure it differently:

```<rdf:Description rdf:about="" xmlns:GPano="http://ns.google.com/photos/1.0/panorama/">
      <GPano:ProjectionType>equirectangular</GPano:ProjectionType>
      <GPano:UsePanoramaViewer>True</GPano:UsePanoramaViewer>
      <GPano:CroppedAreaImageWidthPixels>5376</GPano:CroppedAreaImageWidthPixels>
      <GPano:CroppedAreaImageHeightPixels>2688</GPano:CroppedAreaImageHeightPixels>
      <GPano:FullPanoWidthPixels>5376</GPano:FullPanoWidthPixels>
      <GPano:FullPanoHeightPixels>2688</GPano:FullPanoHeightPixels>
      <GPano:CroppedAreaLeftPixels>0</GPano:CroppedAreaLeftPixels>
      <GPano:CroppedAreaTopPixels>0</GPano:CroppedAreaTopPixels>
      <GPano:PoseHeadingDegrees>270.0</GPano:PoseHeadingDegrees>
      <GPano:PosePitchDegrees>-1.3</GPano:PosePitchDegrees>
      <GPano:PoseRollDegrees>0.2</GPano:PoseRollDegrees>
    </rdf:Description>```

**Comment by mistic100** [comment]
http://stackoverflow.com/questions/32582441/are-these-two-xml-equivalent
TLDR: these syntax are the same for a RDF reader but not for a XML reader.

I'll update the parsing method (the only thing I didn't modified from the original lib).

Could please send me the file which causes the bug ?

**Comment by Davidx7** [comment]
Made a quick fix to make it work, without parsing XML anymore, just extracting when I need.

```var getexiftag = function(data, attr) {
return data.substring(data.indexOf('<GPano:'+attr+'>'),data.indexOf('</GPano:'+attr+'>')).replace('<GPano:'+attr+'>','');
              };```

The photo is a sphere taken with a Google Camera app on a Galaxy S5
Made this hard fix check ```if (isNaN(PSVUtils.getAttribute(data, 'FullPanoWidthPixels'))) {``` and to see if it's your format or the one I found. 

Works.

**Comment by mistic100** [comment]
(when using triple back tilde you must put it a new line otherwise it does not work, I edited your messages)
